libs/providers/vite.config.ts

```diff
@@ -2,16 +2,51 @@
 import { defineConfig } from 'vite';
 import react from '@vitejs/plugin-react';
 import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';
+import dts from 'vite-plugin-dts';
+import * as path from 'path';
 
 export default defineConfig({
   cacheDir: '../../node_modules/.vite/test-vitest',
 
-  plugins: [react(), nxViteTsPaths()],
+  plugins: [
+    dts({
+      entryRoot: 'src',
+      tsConfigFilePath: path.join(__dirname, 'tsconfig.lib.json'),
+      skipDiagnostics: true,
+      // insertTypesEntry: true,
+    }),
+    react(),
+    nxViteTsPaths(),
+  ],
 
   // Uncomment this if you are using workers.
   // worker: {
   //  plugins: [ nxViteTsPaths() ],
   // },
+  build: {
+    lib: {
+      // Could also be a dictionary or array of multiple entry points.
+      entry: 'src/index.ts',
+      name: 'providers',
+      fileName: 'index',
+      // Change this to the formats you want to support.
+      // Don't forget to update your package.json as well.
+      formats: ['es', 'cjs'],
+    },
+    rollupOptions: {
+      // External packages that should not be bundled into your library.
+      external: [
+        'react',
+        'react-dom',
+        'react/jsx-runtime',
+        'tailwind-merge',
+        '@deriv/quill-design',
+        '@deriv/quill-icons',
+        '@deriv-com/components',
+      ],
```

Comment:
is it possible to use **rollup-plugin-peer-deps-external** plugin?

Reply:
not now, @hasan-deriv , since this is a monorepo setup, we have do it like this for now. otherwise we have to add this to package.json and use the plugin. we can do it later if needs be. and thanks for the suggestion 🙏